src/Command/StokeCommand.php

```diff
@@ -498,7 +651,29 @@ protected function pickDisplayName(array $displayNames, array $options)
      */
     private function getFilePathForEntityId($entityId)
     {
-        $filePath = $this->metadataDirectory . md5($entityId) . '.xml';
+		$md5entityID =  md5($entityId);
+		
+        $filePath = $this->metadataDirectory  . $md5entityID . '.xml';
+        return $filePath;
+    }
+    
+    private function getFilePathForLogo($entityId)
+    {
+        $md5entityID =  md5($entityId);
+    
+        // Create subdirs based on the first letter of the md5 of the entityID to avoid filling the dir with too many files
+        // Note the logos are always in the logos subdir
+        $filePath = $this->metadataDirectory. "logos/" . $md5entityID[0] . "/";
+        return $filePath;
+    }
+    
+    private function getURLPathForLogo($entityId)
+    {
+        $md5entityID =  md5($entityId);
```

Comment:
That we're using MD5 is an implementation concern, we're hashing the entityId to make it safe to use on the file system.
Now that it's in three places this might be better in a separate function:
```
private function getFileNameForEntityId($entityId) {
    return md5($entityId);
}
```